### DISCUSSION TITLE
How to compile in Windows?

### DISCUSSION THREAD

**Discussion by Rennngw** [coal-library/coal#470] [Q&A] [answered]
Hello, I tried to compile hpp-fcl on the windows system. I used cmake-gui VS2022 to compile, and the following error occurred. How can I solve this problem?

![image](https://github.com/humanoid-path-planner/hpp-fcl/assets/110526351/3d95512f-a197-4f01-9a10-263604712074)

**Comment by jcarpent** [comment, marked as answer]
You can use conda to install on Windows.
```bash
conda install hpp-fcl -c conda-forge
```

**Reply by Rennngw** [reply]
Thanks for the reply, thanks for solving my problem.

**Comment by jcarpent** [comment]
Do you really need to compile from source?

**Reply by Preston2jager** [reply]
Can I suggest adding the conda installation info to the INSTALL file ? I came across this package from ifcClash as a dependency. While checking the INSTALL file,  I throuhgt I have to build it from source and stuck with some octomap package issues for days. The conda installation just saved my day.

**Comment by JeffHough** [comment]
@jcarpent 

Is there any plans to support `hpp-fcl` install through pip on windows? Or is only the conda install going to be supported for windows moving forward?

As of today, it seems this is not supported:
`pip install hpp-fcl`

The result on my machine is that the `cmeel` fails to build assimp.

I've worked around this temporarily using the conda solution above, however I wasn't using a conda environment before on this project, and I'm not sure if I will be able to use one moving forward given constraints of the particular project I'm working on.

Thanks!

**Reply by nim65s** [reply]
no, windows support is not planned for pip. conda is a solution, building things yourself another, and you could also use WSL.
